### ISSUE TITLE
Dune fire trooper RiPLib issue (one less dmg)

### ISSUE THREAD

**Issue by TheFlamelet** [wesnoth/wesnoth#8273]
### Describe the desired feature

Hello,

1. So I have read through RipL is bad discussions, and there seem to be consensus that violation of RiPL is only allowed when there is a good reason why it should happen.
2. Furthermore, it was mentioned that RiPL often happens due to oversight when buffing lower levels of the unit.
3. Firetrooper RiPL is bad issue has been raised in discord, and no good reason was found. It was advised to open the issue here, and if there no good reason to keep the RiPL, buff the firetrooper to 10x4, making it satisfy the rule.
4. I didn't notice this was important matter until I realized that 1.18 is a stable version and must be compatible with upcoming versions, meaning that if we dont fix this now, we must wait until 1.20.

Tl;Dr 
Dune scorcher buff +dmg caused RiPL. Firetrooper must be buffed unless there is a good reason to keep the RiPL.

Scorcher dmg: 10 × 3
FireTrooper dmg: 9 × 4

**Comment by TheFlamelet** [comment]
Reserved for similar unit dmg, to give readers feeling on the dune fire trooper line.

Orcish Crossbow
Orcish Slurbow

Fire Drake
Inferno Drake
ArmaGGedon Drake

**Comment by CelticMinstrel** [comment]
Why exactly is 9x4 worse than 10x3? It has 1 more strike and greater potential damage.

**Comment by TheFlamelet** [comment]
> Why exactly is 9x4 worse than 10x3? It has 1 more strike and greater potential damage.

any damage reduction can be worse in some cases.

for example, scorcher will have better chance to kill 24 HP skeleton than burner will, and similar cases.

**Comment by Hejnewar** [comment]
So...
Units actually never fully follow RiPLB, the start to cost more upkeep, give more xp to the enemies when fighting or dying. Finally this wouldnt be the only case, skele archer is another one. In general I dont consider such small things to be problematic or worth caring about a lot.

**Comment by TheFlamelet** [comment]
> So... Units actually never fully follow RiPLB, the start to cost more upkeep, give more xp to the enemies when fighting or dying. Finally this wouldnt be the only case, skele archer is another one. 

Xp, upkeep, and cost are not considered RipL becuz they apply to all units (also consensus)

I deliberately avoided bring up other cases of RiPL becuz each case might have their own justification, and the consensus was RiPL violation is acceptable if there is a good justification for it.

Drake clasher line might be one if the best examples for such good reason. We might go into other units too but that will diverge off our issue here.

However,  the only justification explained in our Dune Trooper case, was explained in discord by ( x )
"Becuz Fire Trooper is lvl 3 unit and those lvl 3 units hardly appear on MP and need not to be balanced"

However, this very reasoning can be used to buff the damage from 9=>10 becuz it's a lvl 3 unit.

The question still remains: why do we want scorcher to have better kill chance than his advanced FireTrooper? In the cases of multiples of 12 described in previous comment?

> In general I dont consider such small things to be problematic or worth caring about a lot.

I used to agree with this until I realised that we have to stick with this problem until.... 1.20...

**Comment by TheFlamelet** [comment]
For the record, when I say consensus, im referring to the old threads linked in wesnoth glossary which defined and debated if RiPL is bad or not,
and there were things which were accepted by both sides of the debate.

http://www.wesnoth.org/forum/viewtopic.php?t=7205

https://wiki.wesnoth.org/Glossary

**Comment by Hejnewar** [comment]
What agruments, I just dont think that this is major enough to change for sole reason of RiPLB. 

If anything I would go with 8-5, that should solve it.

**Comment by TheFlamelet** [comment]
First image: vs mashuroomed skeleton
second vs flattenned skeleton
third vs sand skeleton 
fourth vs swimming skeleton
fifth vs submerged skeeton
Finally vs trivial 50% skeleton of hills and woods

![image](https://github.com/wesnoth/wesnoth/assets/155806857/96f4c737-68ee-416b-8679-77d1f9870dfb)
![image](https://github.com/wesnoth/wesnoth/assets/155806857/7cc57a03-867a-4c35-be7a-8d40bb46083b)

![image](https://github.com/wesnoth/wesnoth/assets/155806857/3e977beb-5cd5-41d9-9280-02ea50d69eb8)
![image](https://github.com/wesnoth/wesnoth/assets/155806857/d39750aa-64df-491d-a59c-a44966e76299)

![image](https://github.com/wesnoth/wesnoth/assets/155806857/d35e990f-a618-4397-b58d-ebdeff4f5ba6)
![image](https://github.com/wesnoth/wesnoth/assets/155806857/a764dd38-3f9d-4180-ac9a-b7303702d798)

![image](https://github.com/wesnoth/wesnoth/assets/155806857/dd4913ab-b8e8-4947-830e-a39f8160ce86)
![image](https://github.com/wesnoth/wesnoth/assets/155806857/56249b4b-5c4a-4ccf-9f10-f2e2f71cf564)


![image](https://github.com/wesnoth/wesnoth/assets/155806857/ee235f50-50fc-4fee-a7b6-dee2c5897e5c)
![image](https://github.com/wesnoth/wesnoth/assets/155806857/b825f3fd-3c20-42a6-8665-4b32a5342d87)


![image](https://github.com/wesnoth/wesnoth/assets/155806857/0a961fd2-2e9b-4d05-9cb1-bf5830b4de9c)
![image](https://github.com/wesnoth/wesnoth/assets/155806857/04c1a007-65ad-4f86-a32d-9281f27ef027)

**Comment by Pentarctagon** [comment]
Maybe I've been misunderstanding riplib this whole time, but I always interpreted it to mean that the damage the unit deals overall is greater. Not that both the damage and the number of strikes must both be equal or greater than the base unit.

1a from the thread you linked:
> 1a) A significant decrease in either the number of swings or amount of damage. If the decrease is very slight (e.g. 7-4 -> 6-6) then it can generally be ignored, but if an attack goes from 8-3 to 3-15, it is a problem.

So by that criteria, this unit is fine.

**Comment by TheFlamelet** [comment]
> Maybe I've been misunderstanding riplib this whole time, but I always interpreted it to mean that the damage the unit deals overall is greater. Not that both the damage and the number of strikes must both be equal or greater than the base unit.
>

It will be slight decrease in power. The RiPL tries to catch all aspects, not just on average. From 4th post.  

> Flamelet said: but the question still remains; why do we want scorcher to have better kill chance than his advanced FireTrooper?

> 1a from the thread you linked:
> So by that criteria, this unit is fine.

Iirc not everyone agreed on that. The question still remains.



Also, with regards to the probability of the other proposed solution by Hejnewar of 8x5 instead of 10x4; I interpreted the probability there as strictly better CTK (Chance% to kill) in the mentioned cases so far but I might have to review the data or look for marginal cases

Hint: if you need 1 strike to kill, omit the first probability and add the rest
If you need 2 strikes, ommit the first two and add the rest.
You will see 8x5 is strictly higher than 10x3 to make a kill
Test data was vs skeleton.



8x5 is likeable no. strikes. If proven to be strictly better why not.
